genet/input/gtfs_reader.py

```diff
@@ -179,11 +179,11 @@ def generate_routes(group):
     df['arrival_time'] = df['arrival_time'].apply(lambda x: get_time(x))
     df['departure_time'] = df['departure_time'].apply(lambda x: get_time(x))
 
-    df = df.groupby('trip_id', as_index=False).apply(generate_stop_sequence).reset_index()
+    df = df.groupby('trip_id', as_index=False).apply(generate_stop_sequence).reset_index(drop=True)
     # drop stop sequences that are single stops
     df = df[df['ordered_stops'].str.len() > 1]
     df['vehicle_id'] = [f'veh_{i}' for i in range(len(df))]
-    df = df.groupby(['route_id', 'stops_str'], as_index=False).apply(generate_trips).reset_index()
+    df = df.groupby(['route_id', 'stops_str'], as_index=False).apply(generate_trips).reset_index(drop=True)
```

Comment:
I should have removed the index resetting when I introduced the `as_index=False` groupby arg. If you remove `.reset_index(drop=True)` entirely (inc. on line 182), do you get the same result?